### ISSUE TITLE
Interoperability of indexing-based SIMD interface

### ISSUE THREAD

**Issue by tkf** [eschnett/SIMD.jl#45]
In my small sparse array package https://github.com/tkf/SparseXX.jl, I want to use indexing-based SIMD interface with different "non-standard" arrays like the ones from MappedArrays.jl and FillArrays.jl.  It may even make sense to support SIMD-indexing `Broadcasted` object so that it works with LazyArrays.jl.  At first, I was rather hoping that I can do it by some duck typing so that array libraries do not need to know about SIMD.jl.  However, as pointed out by @timholy and @mbauman in https://github.com/JuliaArrays/MappedArrays.jl/pull/24, it looks we need _some_ code importing both SIMD.jl and the array libraries and define `getindex` etc. that are aware of those libraries.

The question is who distribute/maintain the code.  As I'm the only one using this feature now, I think it's probably better if I create a package that does it (here is the proof-of-concept: https://github.com/tkf/SIMDIndexing.jl).  But then it becomes technically a type piracy since I don't "own" those types or functions.  So, I thought it's better to get the "permission" of doing the type piracy in a friendly fashion.

Of course, if you want to include it in SIMD.jl as "soft" dependencies using `@require`, I will happily create a PR.

cc: @dlfivefifty, as you seems to be maintaining FillArrays.jl

**Comment by dlfivefifty** [comment]
SIMD.jl is a very lightweight package, why not just have FillArrays.jl depend on SIMD.jl?

**Comment by timholy** [comment]
16 exports and almost 2k LOC isn't lightweight compared to some AbstractArray types. For example, PaddedViews is roughly 10% of the size of SIMD.

**Comment by dlfivefifty** [comment]
I meant lightweight in terms of it has no dependencies.

**Comment by tkf** [comment]
> FillArrays.jl depend on SIMD.jl?

Yes, this is a possible solution.  But that's `O(N)` work for me where `N` = number of array package authors because I have to keep convincing them to have SIMD.jl as a dependency or to use Requires.jl.  In practice `N` is probably 2 (`= #{@timholy, @dlfivefifty}`) but the possibility that `N` may grow makes me not wanting to go this direction...
